### PR TITLE
migrate in emboss_5, fastqc, freebayes, picard from devteam

### PR DESCRIPTION
**Pull request by martenson**
part of https://github.com/galaxyproject/tools-devteam/issues/406
xref: https://github.com/galaxyproject/tools-iuc/pull/1060

ping @galaxyproject/iuc 

this is an effort to address part of https://github.com/galaxyproject/tools-devteam/issues/416

### CODE REVIEW

**Comment by martenson** [comment]
thanks to @erasche for helping with figuring out how to treat the missing docs

**Comment by martenson** [comment]
freebayes fails linting because of `<conversion>` not being allowed in schema https://docs.galaxyproject.org/en/latest/dev/schema.html

```
<param name="input_variant_vcf" type="data" format="vcf_bgzip" label="Use variants reported in VCF file as input to the algorithm">
    <conversion name="Tabixized_input" type="tabix" />
</param>
```

**Comment by nsoranzo** [comment]
@martenson I've pushed 2 commits to your branch and I plan to do some more work tomorrow, hope that's OK.

**Comment by yhoogstrate** [comment]
Same for me (Just one pr)

File: tools/picard/read_group_macros.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+../../macros/read_group_macros.xml
```
Comment (nsoranzo) [inline tools/picard/read_group_macros.xml]: I've added the file in commit 10954c0 .

**Comment by nsoranzo** [comment]
I made a massive clean-up of the EMBOSS wrappers in commit cafb44d . More can be done post-merge, e.g.:
- fix datatype assignment to output datasets
- change many `type="select"` params to `boolean`.

**Comment by nsoranzo** [comment]
> freebayes fails linting because of `<conversion>` not being allowed in schema

@martenson @jmchilton I've created https://github.com/galaxyproject/galaxy/issues/3453 to track this.

**Comment by nsoranzo** [comment]
@martenson I've added the changes requested in my review of https://github.com/martenson/tools-iuc/pull/1 .
@jmchilton If you can update Planemo with the new XSD changes, then the tests should pass.

File: tools/emboss_5/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
```
Comment (yhoogstrate) [inline tools/emboss_5/tool_dependencies.xml]: @nsoranzo Can this be removed?

File: tools/fastqc/rgFastQC.xml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<tool id="fastqc" name="FastQC" version="0.67">
+<tool id="fastqc" name="FastQC" version="0.11.5.0">
```
Comment (martenson) [inline tools/fastqc/rgFastQC.xml]: You effectively downgraded the version here.

File: tools/emboss_5/tool_dependencies.xml (hunk 2 above)
Comment (nsoranzo) [inline tools/emboss_5/tool_dependencies.xml]: It is not in Bioconda, but it is in `iuc`:
```
$ conda search emboss
Fetching package metadata ...............
emboss                       5.0.0                         0  iuc             
                             6.5.7                         0  bioconda        
                             6.5.7                         1  bioconda        
                             6.5.7                         2  bioconda        
```
Anyway, I have no strong opinion, maybe we can remove it when we upgrade to a newer EMBOSS version.

**Comment by martenson** [comment]
All tests are failing now, I don't know why.

**Comment by nsoranzo** [comment]
The problem was that release_16.10 has not been update to require six 1.10, but now that we have a `twill` wheel and a new Planemo release (thanks to @jmchilton), the linting passes and only the Java tools fail for same reason. @bgruening ideas?

File: tools/fastqc/rgFastQC.xml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        <requirement type="package" version="0.11.5">fastqc</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+        <exit_code range=":-1" />
+        <regex match="Error:" />
+        <regex match="Exception:" />
+    </stdio>
+    <command><![CDATA[
+        python '${__tool_directory__}/rgFastQC.py'
```
Comment (bgruening) [inline tools/fastqc/rgFastQC.xml]: Nicola this looks wired, isn't it?

Comment (nsoranzo) [inline tools/fastqc/rgFastQC.xml]: Why? rgFastQC.py is part of the wrapper.

Comment (bgruening) [inline tools/fastqc/rgFastQC.xml]: Isn't it a string substitution: https://github.com/galaxyproject/galaxy/blob/29cdd6007fa333e8e9caa85a1fc58ae389405cf2/lib/galaxy/jobs/__init__.py#L983
and should be `$__tool_directory__`

Comment (nsoranzo) [inline tools/fastqc/rgFastQC.xml]: The curly braces should be fine, see https://docs.python.org/2/library/string.html#template-strings

Comment (nsoranzo) [inline tools/fastqc/rgFastQC.xml]: I think I've fixed the FastQC issue with commit d38e1e7, let's see.

Comment (bgruening) [inline tools/fastqc/rgFastQC.xml]: Oh ok! Good to know. I thought thats the reason for the strange error message. I still think we should change the variable name and keep is consistent with other places.
Thanks NIcola!

**Comment by martenson** [comment]
fastqc now passes, yay! thanks @nsoranzo

**Comment by nsoranzo** [comment]
So FastQC is finally fixed, but all Picard tools still fail because of this error: Exception in thread "main" java.lang.UnsupportedClassVersionError: picard/cmdline/PicardCommandLine : Unsupported major.minor version 52.0

This indicates that Picard tools are run with Java <8 even if the correct java-jdk Bioconda package is installed. Maybe we need to use Galaxy 17.01? @bgruening

**Comment by nsoranzo** [comment]
The JAVA_HOME change was not applied during the Travis build, trying to close/reopen.

**Comment by bgruening** [comment]
Thanks @nsoranzo. Thats seems to fix it.
Ready for the merge?

**Comment by nsoranzo** [comment]
Green at last! Thanks all!

**Comment by martenson** [comment]
Splendid. Thanks @nsoranzo @bgruening and @yhoogstrate for help!

**Pull request merged**